specification/servicenetworking/resource-manager/Microsoft.ServiceNetworking/preview/2024-05-01-preview/TrafficController.json

```diff
@@ -0,0 +1,1895 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "TrafficController",
+    "version": "2024-05-01-preview",
+    "description": "Traffic Controller Provider management API.",
+    "x-typespec-generated": [
+      {
+        "emitter": "@azure-tools/typespec-autorest"
+      }
+    ]
+  },
+  "schemes": [
+    "https"
+  ],
+  "host": "management.azure.com",
+  "produces": [
+    "application/json"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "description": "Azure Active Directory OAuth2 Flow.",
+      "flow": "implicit",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "tags": [
+    {
+      "name": "AssociationsInterface"
+    },
+    {
+      "name": "FrontendsInterface"
+    },
+    {
+      "name": "SecurityPoliciesInterface"
+    },
+    {
+      "name": "TrafficControllerInterface"
+    },
+    {
+      "name": "Operations"
+    }
+  ],
+  "paths": {
+    "/providers/Microsoft.ServiceNetworking/operations": {
+      "get": {
+        "operationId": "Operations_List",
+        "tags": [
+          "Operations"
+        ],
+        "description": "List the operations for the provider",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
```

Comment:
>v3 [](http://example.com/codeflow?start=69&length=2)

Here and everywhere else in the file : Why cant v5 be used here ?

Reply:
changing it to v5 throws several validation errors. I had made the change but I had to revert it.

Reply:
Please see if you can resolve the validation errors individually, rather than just avoid v5 entirely because of it.

Reply:
validation errors were resolved. Changed to v5

Reply:
moving to v5 causes breaking changes. I'm unsure of how to only partially specific version of the common types.

Reply:
Please recommend how breaking changes are to be handled or the syntax for multiple versions in the typespec. With the back and forth of validation errors, it is extremely difficult to incorporate suggested changes.